lib/playlistGenerator/PlaylistGenerator.js

```diff
@@ -267,7 +275,8 @@ var doPersist =  function() {
 };
 
 // save playlist
-var persistPlayList = function persistPlayList(force){
+// todo: if stream ended set presentationEndTime to now - playWindow
+var persistPlayList = function persistPlayList(force, ended = false){
```

Comment:
it's a bit combersome... i think it'd be better if you expose an interface for the playlist generator that you will setup if it ended or now